numpy/polynomial/chebyshev.py

```diff
@@ -1594,6 +1700,70 @@ def chebvander3d(x, y, z, deg):
     return v.reshape(v.shape[:-3] + (-1,))
 
 
+def chebvandernd(deg, *xi):
+    """Pseudo-Vandermonde matrix of given degrees.
+
+    Returns the pseudo-Vandermonde matrix of degrees `deg` and sample
+    points `(x1, x2, ..., xn)`. If `l, m, n` are the given degrees in
+    `x1, x2, x3`, then The pseudo-Vandermonde matrix is defined by
+
+    .. math:: V[..., (m+1)(n+1)i + (n+1)j + k] = T_i(x1)*T_j(x2)*T_k(x3),
+
+    where `0 <= i <= l`, `0 <= j <= m`, and `0 <= k <= n`.  The leading
+    indices of `V` index the points `(x, y, z)` and the last index encodes
+    the degrees of the Chebyshev polynomials.
+
+    If ``V = chebvandernd([xdeg, ydeg, zdeg], x, y, z)``, then the columns
+    of `V` correspond to the elements of a 3-D coefficient array `c` of
+    shape (xdeg + 1, ydeg + 1, zdeg + 1) in the order
+
+    .. math:: c_{000}, c_{001}, c_{002},... , c_{010}, c_{011}, c_{012},...
+
+    and ``np.dot(V, c.flat)`` and ``chebvalnd(c, x, y, z)`` will be the
+    same up to roundoff. This equivalence is useful both for least squares
+    fitting and for the evaluation of a large number of N-D Chebyshev
+    series of the same degrees and sample points.
+
+    Parameters
+    ----------
+    deg : list of ints
+        List of maximum degrees of the form [x1_deg, x2_deg, ...,xn_deg].
+    x1, x2, ..., xn : array_like
+        Arrays of point coordinates, all of the same shape. The dtypes will
+        be converted to either float64 or complex128 depending on whether
+        any of the elements are complex. Scalars are converted to 1-D
+        arrays.
+
+    Returns
+    -------
+    vander : ndarray
+        The shape of the returned matrix is ``x1.shape + (order,)``, where
+        :math:`order = (deg[0]+1)*(deg([1]+1)*...*(deg[n-1]+1)`.  The dtype will
+        be the same as the converted `x1`, `x2`, ... `xn`.
+
+    See Also
+    --------
+    chebvander, chebvalnd, chebfitnd
+    """
+    ideg = [int(d) for d in deg]
+    is_valid = np.array([id == d and id >= 0 for id, d in zip(ideg, deg)])
+    if np.any(is_valid != 1):
+        raise ValueError("degrees must be non-negative integers")
+    ndim = len(xi)
+    if len(ideg)!=ndim:
+        raise ValueError('length of deg must be the same as number of dimensions')
+
+    xi = np.array(xi, copy=0) + 0.0
+    shape0 = xi[0].shape
+    s0 = (1,) * ndim
+    vxi = [chebvander(x, d).reshape(shape0 + s0[:i] + (-1,) + s0[i + 1::])
```

Comment:
This might read better as a for loop with several lines rather than a list comprehension.